### ISSUE TITLE
Redundant Mul ops in Yolov8 TFLite export

### ISSUE THREAD

**Issue by Saurya0401** [ultralytics/ultralytics#21477]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Export

### Bug

Models exported with `model.export(format="tflite", imgsz=...)` have seemingly redundant Mul ops in the convolution blocks. Specifically, the inputs to the gating `Mul` op are multiplied by -1 twice in succession, effectively resulting in an identity operation.
These `Mul` ops do not appear in the FP16 TFLite model or if `onnx2tf` is used to re-export the intermediary ONNX file. The issue seems to be specific to the direct TFLite FP32 export path. Here's a visual comparison of the `onnx2tf` and `model.export()` FP32 models:

![Image](https://github.com/user-attachments/assets/7af88145-7ef7-46c9-8ce9-9a1602e26250)

These operations have negligible impact on inference performance for CPU/GPU targets. However, they are problematic for embedded runtimes with less capable FP32 execution units. Would it be possible to clean up these ops during export or as part of a post-processing step?

### Environment

```
Ultralytics 8.3.168 🚀 Python-3.10.17 torch-2.7.1+cu126 CPU (12th Gen Intel Core(TM) i7-1265U)
Setup complete ✅ (12 CPUs, 15.3 GB RAM, 242.1/1006.9 GB disk)

OS                  Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.39
Environment         Linux
Python              3.10.17
Install             pip
Path                ~/yolov8-test/.venv/lib/python3.10/site-packages/ultralytics
RAM                 15.34 GB
Disk                242.1/1006.9 GB
CPU                 12th Gen Intel Core(TM) i7-1265U
CPU count           12
GPU                 None
GPU count           None
CUDA                None

numpy               ✅ 1.26.4>=1.23.0
matplotlib          ✅ 3.10.3>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 11.3.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.4>=2.23.0
scipy               ✅ 1.15.3>=1.4.1
torch               ✅ 2.7.1>=1.8.0
torch               ✅ 2.7.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.22.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.3.1>=1.1.4
ultralytics-thop    ✅ 2.0.14>=2.0.0
```

### Minimal Reproducible Example

Yolov8 export code:
```py
# export.py

from ultralytics import YOLO

model = YOLO("yolov8s.pt")
model.export(format="tflite", imgsz=(384, 640))
```

Direct TFLite export:
```sh
python export.py
```

`onnx2tf` re-export:
```sh
python export.py
onnx2tf -i yolov8s.onnx -o yolov8s_manual
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Saurya0401, thank you for your detailed report and interest in Ultralytics 🚀! We appreciate you highlighting the redundant Mul ops seen in TFLite exports and sharing your thorough environment details and visual comparison.

This is an automated response to help you get started. An Ultralytics engineer will review your issue and assist you soon 🛠️.

If this is a 🐛 Bug Report, please ensure you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/), which you have done—thank you! This will help us investigate and resolve the issue more efficiently.

If you have additional information, logs, or model files that could further clarify the behavior, please feel free to share them.

For more information and guidance, check out the [Docs](https://docs.ultralytics.com/), including [Python usage](https://docs.ultralytics.com/usage/python/) and [CLI examples](https://docs.ultralytics.com/usage/cli/). You can also connect with the Ultralytics community for real-time help on [Discord](https://discord.com/invite/ultralytics) 🎧, join deeper discussions on [Discourse](https://community.ultralytics.com/), or visit our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

To ensure this issue hasn't already been resolved, please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your report! Your input helps us improve the Ultralytics ecosystem.

**Comment by Y-T-G** [comment]
It's due to this:

https://github.com/ultralytics/ultralytics/blob/9dccbd42298ce801054cb90cd0da0442b2adb10d/ultralytics/engine/exporter.py#L1017

I am not sure if it's intended or a bug in onnx2tf.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in PR #21482. This update removes the unused optimization_for_gpu_delegate flag from the TensorFlow SavedModel export process, simplifying the workflow and potentially improving compatibility for those exporting models to this format. The PR also includes a small documentation update to keep contributor information current.

Thanks to the efforts of the PR author and contributors, these changes should address the concerns reported here.

To test if this resolves your issue, try installing the latest code from the main branch using:

pip install git+https://github.com/ultralytics/ultralytics.git@main

Alternatively, you may wait for these changes to be included in the next official PyPI release.

Please let us know if this update resolves your issue, or if any problems persist. Thank you for reporting this and helping improve Ultralytics! Your feedback is always appreciated.
